schemachanger: implement RENAME COLUMN in the declarative schema changer

rafiss opened the pull request:
This was a pretty straightforward copy of the logic from the legacy schema changer.

Only one new dependency rule was needed, to make sure the old column name is removed
before the new one is added.

fixes https://github.com/cockroachdb/cockroach/issues/148340
fixes #139840
Release note: None

annrpom committed:
schemachanger: implement RENAME COLUMN in the declarative schema changer

Release note: None

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

@rafiss @annrpom  Nice work, just a few minor comments

> @fqazi reviewed 23 of 23 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152769#-:-OZ9zE1R-GUJJckxwVyz:baaxs98)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rafiss)

___
*[`pkg/sql/schemachanger/scplan/internal/rules/current/dep_rename_column.go` line 27 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/152769#-OZ9ymYO53DUgXgcECf4:-OZ9ymYO53DUgXgcECf5:b5luk2z) ([raw file](https://github.com/cockroachdb/cockroach/blob/390abaffa9cdd0d7206a1e5ad9432b1c5ac6acf9/pkg/sql/schemachanger/scplan/internal/rules/current/dep_rename_column.go#L27)):*
> ```Go
> 				from.Type((*scpb.ColumnName)(nil)),
> 				to.Type((*scpb.ColumnName)(nil)),
> 				from.El.AttrEqVar(rel.Type, "same-type"),
> ```

We only have a single type above so this isn't needed.
___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_rename_column.go` line 47 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/152769#-OZ9xv8S2mjExoBVhqn_:-OZ9xv8S2mjExoBVhqna:btpi8k4) ([raw file](https://github.com/cockroachdb/cockroach/blob/390abaffa9cdd0d7206a1e5ad9432b1c5ac6acf9/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_rename_column.go#L47)):*
> ```Go
> 	}
> 
> 	if colNameElt.Name == string(t.NewName) {
> ```

Do we care about checks about system columns, etc in the next line?


<!-- Sent from Reviewable.io -->

rafiss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152769)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @fqazi)

___
*[`pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_rename_column.go` line 47 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/152769#-OZ9xv8S2mjExoBVhqn_:-OZAoXSO8Ln4QZppuDFA:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/390abaffa9cdd0d7206a1e5ad9432b1c5ac6acf9/pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_rename_column.go#L47)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Do we care about checks about system columns, etc in the next line?
</blockquote></details>

done
___
*[`pkg/sql/schemachanger/scplan/internal/rules/current/dep_rename_column.go` line 27 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/152769#-OZ9ymYO53DUgXgcECf4:-OZAC1NI-moZ6GXvuokF:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/390abaffa9cdd0d7206a1e5ad9432b1c5ac6acf9/pkg/sql/schemachanger/scplan/internal/rules/current/dep_rename_column.go#L27)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

We only have a single type above so this isn't needed.
</blockquote></details>

done


<!-- Sent from Reviewable.io -->

rafiss committed:
logictest: restore tests that were commented a long time ago

Release note: None

rafiss committed:
scbuild: change parameter type of helper functions

This makes it easier to remember to format the name correctly.

This also adjusts an error message to be slightly more accurate.

Release note: None

rafiss commented:
tftr!

bors r+
